boards/sipeed-longan-nano/Makefile.dep

```diff
@@ -12,8 +12,15 @@ ifneq (,$(filter vfs_default,$(USEMODULE)))
   USEMODULE += mtd
 endif
 
-ifneq (,$(filter disp_dev,$(USEMODULE)))
-  USEMODULE += st7735
+ifneq (,$(filter sipeed_longan_nano_tft,$(USEMODULE)))
+  ifneq (,$(filter disp_dev,$(USEMODULE)))
+    USEMODULE += st77xx
+  endif
+
+  ifneq (,$(filter st77xx,$(USEMODULE)))
+    # indicate that a ST7735 is used
+    USEMODULE += st7735
+  endif
```

Comment:
Modeling consistent dependencies in Kconfig and the makefiles are really a pain :worried: It took me a number of iteration cycles to get the dependencies as compiled in  the CI working. First, I did it with the definition of such a `DRIVER` variable and the driver variants as you suggested, but I couldn't get working in that way.

Specifying the variant only if `disp_dev` is enabled but not if just the driver `st77xx` is enabled will cause inconsistencies in `tests/drivers/st77xx`. Kconfig pulls in the driver by `CONFIG_MODULE_ST77XX=y` in `app.config.test` and the `HAVE_ST*` symbols then pull in the specific variant that a given board has. The Makefile dependencies however would always use `st7735` by default because of the `DRIVER` variable. To compile it with the other variants in the CI would then require
- either to separate `tests/driver/st77*` test apps for each variant to be able to set the `DRIVER ?= st77*` variable and `CONFIG_MODULE_ST77*=y` (however the code duplication make definitely no sense)

- or to set the `DRIVER` variable explicitly for all boards that use a certain variant other than the default, for example:
   ```pyhton
   # define the driver to be used for selected boards
   ifneq (,$(filter stm32l496g-disco esp32s3-usb-otg esp32s2-lilygo-ttgo-t8,$(BOARD)))
     DRIVER := st7789
   endif
   
   ifneq (,$(filter esp32s3-wt32-sc01-plus,$(BOARD)))
     DRIVER := l3g4200d_ng
   endif
   ```

To be honest, isn't it more logical that if an app says "I select the display driver `st77xx` whatever variant is used" that the board says "Ok, I have this variant" or it uses the default. This corresponds to the approach as it is modeled in Kconfig.